opm/common/quad.hpp

```diff
@@ -0,0 +1,325 @@
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_pod<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_signed<quad>
+    : public integral_constant<bool, true>
+{};
+
+
+template <>
+struct is_standard_layout<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_trivial<quad>
+    : public integral_constant<bool, true>
+{};
+
+/*
+template <>
+struct is_trivially_copyable<quad>
+    : public integral_constant<bool, true>
+{};
+*/
+
+template <class OtherType>
+struct is_assignable<quad, OtherType>
+    : public integral_constant<bool, is_arithmetic<OtherType>::value>
+{};
+
+template <class OtherType>
+struct is_nothrow_assignable<quad, OtherType>
+    : public is_assignable<quad, OtherType>
+{};
+
+/*
+template <class OtherType>
+struct is_trivially_assignable<quad, OtherType>
+    : public integral_constant<bool, is_arithmetic<OtherType>::value>
+{};
+*/
+
+template <>
+struct is_copy_assignable<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_nothrow_copy_assignable<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_move_assignable<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_nothrow_move_assignable<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_constructible<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_nothrow_constructible<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_default_constructible<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_nothrow_default_constructible<quad>
+    : public integral_constant<bool, true>
+{};
+
+/*
+template <>
+struct is_trivially_default_constructible<quad>
+    : public integral_constant<bool, true>
+{};
+*/
+
+template <>
+struct is_copy_constructible<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_move_constructible<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_nothrow_move_constructible<quad>
+    : public integral_constant<bool, true>
+{};
+
+
+template <>
+struct is_destructible<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <>
+struct is_nothrow_destructible<quad>
+    : public integral_constant<bool, true>
+{};
+
+template <class OtherType>
+struct is_convertible<quad, OtherType>
+    : public is_arithmetic<OtherType>
+{ };
+
+inline std::ostream& operator<<(std::ostream& os, const quad& val)
+{ return (os << double(val)); }
+
+inline std::istream& operator>>(std::istream& is, quad& val)
+{
+    double tmp;
+    std::istream& ret = (is >> tmp);
+    val = tmp;
+    return ret;
+}
+
+inline quad abs(quad val)
+{ return (val < 0) ? -val : val; }
+
+inline quad floor(quad val)
+{ return floorq(val); }
+
+inline quad ceil(quad val)
+{ return ceilq(val); }
+
+inline quad max(quad a, quad b)
```

Comment:
not entirely sure, but I think I added these two because I got compiler errors. will re-check...